### DISCUSSION TITLE
Form partial validation client side

### DISCUSSION THREAD

**Discussion by N00nDay** [huntabyte/shadcn-svelte#523] [Help]
I have a large Form component wired up and I love it. The only issue I have is that I have broken up the form into 4 sections/pages due to its length and I would like to validate the page before I let the user move to the next page. Are there any examples of how to do this using the Form component? I know how to do this manually which will take a lot of boilerplate language to pull off. I am wondering if the component has a way to do this that will simplify the process.

**Comment by babakfp** [comment]
Hey

I don't know about the form component, for form validation, for one of my projects I used [Zod](https://github.com/colinhacks/zod) + [Superforms](https://superforms.rocks/). Maybe it works for your project too. Note that superforms don't work properly for inputs like file input. So, for things like that you need to do it manually.

**Comment by N00nDay** [comment]
I ended up passing `formValues` & `validate` through the "Next" button and processing it myself. Not sure if it is the simplest way of accomplishing the task but it works.
